### PR TITLE
StoredDict is alternative to  PersistentDict

### LINKED ISSUE #14: Consider alternatives to PersistentDict
Use of bluesky's PersistentDict has been problematic at times, as described in these GitHub issues:

- https://github.com/BCDA-APS/bluesky_training/issues/300
- https://github.com/bluesky/bluesky/issues/1741

Consider alternatives to this file-backed Python dictionary, which responsibility is to remember certain metadata keys (in `RE.md`) between sessions, such as `scan_id`.

### PR DESCRIPTION
**Pull request by prjemian**
- close #14

@qzhang234 : Your bug report pushed this along!

### CODE REVIEW

File: src/instrument/configs/iconfig.yml [hunk 1]
```diff
@@ -20,9 +20,12 @@ RUN_ENGINE:
     ### Default: `RE.md["scan_id"]` (not using an EPICS PV)
     # SCAN_ID_PV: f"{IOC}bluesky_scan_id"
 
-    ### Directory to "autosave" the RE.md dictionary (uses PersistentDict).
-    ### Default: HOME/.config/Bluesky_RunEngine_md'
-    # MD_PATH: /home/beams/USERNAME/.config/Bluesky_RunEngine_md
+    ### Where to "autosave" the RE.md dictionary.
+    ### Defaults:
+    # MD_STORAGE_HANDLER: PersistentDict
+    # MD_PATH: HOME/.config/Bluesky_RunEngine_md'
```
Comment (MDecarabas) [inline src/instrument/configs/iconfig.yml:26]: let's go ahead and remove this if we don't need it

File: src/instrument/tests/test_stored_dict.py
Comment (MDecarabas) [inline src/instrument/tests/test_stored_dict.py:1]: If this is just a test file should it not live inside the tests?

File: src/instrument/core/run_engine_init.py [hunk 2]
```diff
@@ -31,13 +32,20 @@
 
 # Save/restore RE.md dictionary, in this precise order.
 if MD_PATH is not None:
+    handler_name = re_config.get("MD_STORAGE_HANDLER", "PersistentDict")
     try:
-        RE.md = bluesky.utils.PersistentDict(MD_PATH)
-    except Exception as e:
+        handler = {
+            "PersistentDict": bluesky.utils.PersistentDict,
+            "StoredDict": StoredDict,
```
Comment (MDecarabas) [inline src/instrument/core/run_engine_init.py:39]: Seems like we are instantiating the StoredDict class exclusively in this dictionary, however at no point are we passing arguments or using the class to do anything. Am I missing something

**Review by MDecarabas — CHANGES_REQUESTED** [review]

File: src/instrument/tests/test_stored_dict.py
Comment (prjemian) [reply src/instrument/tests/test_stored_dict.py:1]: Looks like it's there now.

File: src/instrument/core/run_engine_init.py (hunk 2 above)
Comment (prjemian) [reply src/instrument/core/run_engine_init.py:39]: Here, the StoredDict replaces the default `RE.md` structure: https://github.com/BCDA-APS/bs_model_instrument/blob/54a4d2c8b22f3572470559b184e0003be69cbff1/src/instrument/core/run_engine_init.py#L41

File: src/instrument/configs/iconfig.yml (hunk 1 above)
Comment (prjemian) [reply src/instrument/configs/iconfig.yml]: see commit 96657d2774

File: src/instrument/core/run_engine_init.py (hunk 2 above)
Comment (prjemian) [reply src/instrument/core/run_engine_init.py]: This commit 96657d2774 makes this more obvious.

**Comment by prjemian** [comment]
@MDecarabas All your review comments are addressed.  Please review again.

**Review by MDecarabas — APPROVED** [review]

**Comment by prjemian** [comment]
See https://github.com/bluesky/bluesky/pull/1861

**Pull request merged**